src/main/java/net/goldiriath/plugin/game/questing/quest/triggering/trigger/NPCTrigger.java

```diff
@@ -0,0 +1,43 @@
+package net.goldiriath.plugin.game.questing.quest.triggering.trigger;
+
+import net.citizensnpcs.api.CitizensAPI;
+import net.citizensnpcs.api.npc.NPC;
+import net.goldiriath.plugin.game.questing.quest.triggering.TriggerSource;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.player.PlayerMoveEvent;
+
+import java.util.Objects;
+
+public class NPCTrigger extends AbstractTrigger {
+
+
+    private final int npcID;
+    private final int radiusSquared;
+
+    public NPCTrigger(TriggerSource source, String[] args) {
+        super(source);
+        npcID = parseInt(args[1]);
+        radiusSquared = parseInt(args[2]);
```

Comment:
Why put the squared radius as an argument.  Instead, just multiply the parsed value by itself to keep the source more understandable.

Reply:
Changed in 83cdb0990f1e3e84b25863b3a4d9ae09d03f2a89